infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/receive/PersistedWriterMetadata.java

```diff
@@ -0,0 +1,268 @@
+                    " persisteSiteConfigID " + persisteSiteConfigID + " persistSnapStart " + persistSnapStart);
+            return;
+        }
+
+        TxBuilder txBuilder = corfuStore.tx(namespace);
+
+        //Update the siteConfigID to fence all other transactions that update the metadata at the same time
+        appendUpdate(txBuilder, PersistedWriterMetadataType.SiteConfigID, siteConfigID);
+
+        //Setup the LastSnapStart
+        appendUpdate(txBuilder, PersistedWriterMetadataType.LastSnapTransferDone, ts);
+
+        txBuilder.commit(timestamp);
+
+        log.debug("Commit. Set snapshotStart siteConfigID " + siteConfigID + " ts " + ts +
+                " persisteSiteConfigID " + persisteSiteConfigID + " persistSnapStart " + persistSnapStart);
+        return;
+    }
+
+    public void setSrcBaseSnapshotDone(LogReplicationEntry entry) {
+        CorfuStoreMetadata.Timestamp timestamp = corfuStore.getTimestamp();
+        long persistSiteConfigID = query(timestamp, PersistedWriterMetadataType.SiteConfigID);
+        long persistSnapStart = query(timestamp, PersistedWriterMetadataType.LastSnapStart);
+        long persistSnapTranferDone = query(timestamp, PersistedWriterMetadataType.LastSnapTransferDone);
+        long siteConfigID = entry.getMetadata().getSiteConfigID();
+        long ts = entry.getMetadata().getSnapshotTimestamp();
+
+        if (siteConfigID != persistSiteConfigID || ts != persistSnapStart || ts != persistSnapTranferDone) {
+            log.warn("siteConfigID " + siteConfigID + " != " + " persist " + persistSiteConfigID +  " ts " + ts +
+                    " != " + "persistSnapTranferDone " + persistSnapTranferDone);
+            return;
+        }
+
+        TxBuilder txBuilder = corfuStore.tx(namespace);
+
+        //Update the siteConfigID to fence all other transactions that update the metadata at the same time
+        appendUpdate(txBuilder, PersistedWriterMetadataType.SiteConfigID, siteConfigID);
+
+        appendUpdate(txBuilder, PersistedWriterMetadataType.LastSnapApplyDone, ts);
+        appendUpdate(txBuilder, PersistedWriterMetadataType.LastLogProcessed, ts);
+
+        //may not need
+        appendUpdate(txBuilder, PersistedWriterMetadataType.LastSnapSeqNum, Address.NON_ADDRESS);
+
+        txBuilder.commit(timestamp);
+
+        log.debug("Commit. Set snapshotStart siteConfigID " + siteConfigID + " ts " + ts +
+                " persistSiteConfigID " + persistSiteConfigID + " persistSnapStart " + persistSnapStart);
+
+        return;
+    }
+
+    public static String getPersistedWriterMetadataTableName(UUID primarySite, UUID dst) {
+        return TABLE_PREFIX_NAME + primarySite.toString() + "-to-" + dst.toString();
+    }
+
+    public enum PersistedWriterMetadataType {
+        SiteConfigID("SiteConfigID"),
+        LastSnapStart("lastSnapStart"),
+        LastSnapTransferDone("lastSnapTransferDone"),
+        LastSnapApplyDone("lastSnapApplied"),
+        LastSnapSeqNum("lastSnapSeqNum"),
```

Comment:
could you write a short description for each of these as that will help in understanding the code.